packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -526,6 +643,96 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
   }
 }
 
+/// Positions the options view.
+class _RawAutocompleteOptionsLayoutDelegate extends SingleChildLayoutDelegate {
+  _RawAutocompleteOptionsLayoutDelegate({
+    required this.fieldSize,
+    required this.fieldOffset,
+    required this.optionsViewOpenDirection,
+    required TextDirection? textDirection,
+  }) : textDirection = textDirection ?? TextDirection.ltr;
+
+  /// The size of the field in [RawAutocomplete.fieldViewBuilder].
+  final Size? fieldSize;
+
+  /// The position of the field in [RawAutocomplete.fieldViewBuilder].
+  final Offset? fieldOffset;
+
+   /// A direction in which to open the options view overlay.
+  final OptionsViewOpenDirection optionsViewOpenDirection;
+
+  /// The [TextDirection] of this part of the widget tree.
+  final TextDirection textDirection;
+
+  // A big enough height for about one item in the default
+  // Autocomplete.optionsViewBuilder. The assumption is that the user likely
+  // wants the list of options to move to stay on the screen rather than get any
+  // smaller than this. Allows Autocomplete to work when it has very little
+  // screen height available (as in b/317115348) by positioning itself on top of
+  // the field, while in other cases to size itself based on the height under
+  // the field.
+  static const double _kMinUsableHeight = kMinInteractiveDimension;
+
+  // Limits the child to the space above/below the field, with a minimum, and
+  // with the same maxWidth constraint as the field has.
+  @override
+  BoxConstraints getConstraintsForChild(BoxConstraints constraints) {
+    // shouldRelayout prevents this from being called if fieldOffset or
+    // fieldSize are invalid.
+    assert(fieldSize != null && fieldSize!.isFinite);
+    assert(fieldOffset != null && fieldOffset!.isFinite);
+
+    return BoxConstraints(
+      // The field width may be zero if this is a split RawAutocomplete with no
+      // field of its own. In that case, don't change the constraints width.
+      maxWidth: fieldSize!.width == 0.0 ? constraints.maxWidth : fieldSize!.width,
+      maxHeight: max(
+        _kMinUsableHeight,
+        switch (optionsViewOpenDirection) {
+          OptionsViewOpenDirection.down => constraints.maxHeight - fieldOffset!.dy - fieldSize!.height,
+          OptionsViewOpenDirection.up => fieldOffset!.dy,
+        },
+      ),
+    );
+  }
+
+  // Positions the child above/below the field and aligned with the left/right
+  // side based on text direction.
+  @override
+  Offset getPositionForChild(Size size, Size childSize) {
+    // shouldRelayout prevents this from being called if fieldOffset or
+    // fieldSize are invalid.
+    assert(fieldSize != null && fieldSize!.isFinite);
+    assert(fieldOffset != null && fieldOffset!.isFinite);
```

Comment:
Consider checking this in the constructor.

Reply:
Also why are they types nullable when there are non-null runtime checks everywhere?